kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala

```diff
@@ -64,4 +64,21 @@ object ThreadUtils extends Logging {
         throw new KyuubiException("Exception thrown in awaitResult: ", e)
     }
   }
+
+  def shutdown(
+      executor: ExecutorService,
+      gracePeriod: Duration = FiniteDuration(30, TimeUnit.SECONDS)): Unit = {
```

Comment:
```
gracePeriod: Option[Duration] = None
```

So that:
```
gracePeriod.map { shutdownTimeout =>
executor.awaitTermination(shutdownTimeout, TimeUnit.MILLISECONDS)
}.getOrElse(executor.shutdown())
```

Reply:
This is the refactor for original code in SessionManager.scala#stop().
And I also refered to the ThreadUtils.scala#shutdown in spark code, which has the default 30s for gracePeriod.

To shut down the ExecutorService, we use shutdown() method combined with awaitTermination() method. With this approach, the ExecutorService will first stop taking new tasks and then wait up to a specified period of time for all tasks to be completed. If that time expires, the execution is stopped immediately.